apps/aecore/src/aec_peers_pool.erl

```diff
@@ -0,0 +1,1695 @@
+    undefined -> {undefined, undefined};
+    Peer ->
+      IsAvailable = not (peer_is_selected(Peer) or maps:is_key(PeerId, Standby)),
+      {peer_state(Peer), IsAvailable}
+  end.
+
+%% @doc Returns if the given peer identifier is pooled as verified.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_verified(state(), peer_id()) -> undefined | boolean().
+is_verified(St, PeerId) ->
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> peer_state(Peer) =:= verified
+  end.
+
+%% @doc Returns if the given peer identifier is pooled as unverified.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_unverified(state(), peer_id()) -> undefined | boolean().
+is_unverified(St, PeerId) ->
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> peer_state(Peer) =:= unverified
+  end.
+
+%% @doc Returns if the given peer identifier is available for selection.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_available(state(), peer_id()) -> undefined | boolean().
+is_available(St, PeerId) ->
+  ?ST{standby = Standby} = St,
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> not (peer_is_selected(Peer) or maps:is_key(PeerId, Standby))
+  end.
+
+%% @doc Updates/inserts a peer.
+%%
+%% Takes a time in millisecond as it would be returned from
+%% `erlang:system_time(millisecond)` and uses it as if it was the current time.
+%% Every calls should be given a time greater or equal to the last call.
+%%
+%% If it is the first time this peer is seen, it is added to the verified
+%% pool if trusted, or to the unverified pool otherwise.
+%%
+%% If the peer is already pooled but its address changed, the update is ignored;
+%% this is so the pool cannot get poisoned by nodes gossiping wrong addresses
+%% for valid peers. If a peer rightfully changed its address, the old one will
+%% get removed after the standard retry procedure; at that point, the new one
+%% will be added. Note that this could take quite some time.
+%%
+%% If the peer is already in the verified pool, nothing is done beside updating
+%% the peer last update time and source address.
+%%
+%% If the peer is already in the unverified pool but has not reached the maximum
+%% number of references, another reference may be added to the unverified pool;
+%% the peer last update time and source address fields are updated too.
+%%
+%% Some opaque term can be specified; this term will be returned by `select/4`
+%% alongside the peer identifier. This could be used by the caller to store
+%% extra connection information like protocol and port number. This extra
+%% information will **NOT** be updated if the peer already exists.
+-spec update(state(), millitimestamp(), peer_id(),
+             peer_addr(), peer_addr(), boolean(), extra())
+  -> {verified | unverified | ignored, state()}.
+update(St, Now, PeerId, PeerAddr, SourceAddr, IsTrusted, Extra) ->
+  ?assertNotEqual(undefined, PeerId),
```

Comment:
This is defensive programming and should not be needed. If we don't trust internal data, maybe use a guard, but it really shouldn't be needed either.

Reply:
Asserts are goooooood :) it doesn't exist in production code.

Reply:
If they don't exist in production, they shouldn't exist in the code. Someone might get a false impression that they will actually be used!

Also, I don't see any disabling of asserts so they will be run in production.